mitodl/mitxpro
# feat: catalog topics dropdown

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes https://github.com/mitodl/mitxpro/issues/2614

#### What's this PR do?
Adds a dropdown for the course catalog topics in the Header. The Dropdown also includes links to the main catalog page and programs tab.

#### How should this be manually tested?

- Verify that the Catalog dropdown has all the parent topics and counts.
- Verify that the counts are correct.
- Verify that the Topic links are redirecting to the catalog page with appropriate filters.
- Verify that the `All Topics` link directs to the main catalog page.
- Verify that the `Programs` link directs to the Catalog page with the active `Programs` tab.
- Verify that the number of courses and course count is the same.


#### Where should the reviewer start?
Review the code and then test the changes if there are no or minor review changes.

#### Any background context you want to provide?

- After having a discussion, we decided to keep the design the same as the profile dropdown.
- To avoid extra DB queries for the programs count, we decided to remove the program count for the `Programs` dropdown item. 

#### Screenshots (if appropriate)
(Optional)
<img width="1155" alt="Screen Shot 2023-05-09 at 1 09 05 PM" src="https://user-images.githubusercontent.com/52656433/237035206-a5ee1ee5-ccaf-4a73-b066-c315b801dbe6.png">
<img width="1026" alt="Screen Shot 2023-05-09 at 1 09 19 PM" src="https://user-images.githubusercontent.com/52656433/237035254-7d4f4ac5-2e19-4972-8dbc-739f7822f5e2.png">
<img width="795" alt="Screen Shot 2023-05-09 at 1 09 30 PM" src="https://user-images.githubusercontent.com/52656433/237035261-81b33d4b-bb9c-4e9e-8dcb-089d55148560.png">
<img width="427" alt="Screen Shot 2023-05-09 at 1 09 50 PM" src="https://user-images.githubusercontent.com/52656433/237035267-f1f7ddd8-978c-47de-a3d9-061e22a1eeb4.png">
